File: Saenke Slagskib 22032020 C++/Ship.cpp
#include "Ship.h"
using namespace Game;

Ship::Ship(std::string name, int size, int fieldValue)
	: name(name), size(size), fieldValue(fieldValue)
{
}

std::string Game::Ship::GetName()
{
	return name;
}
int Game::Ship::GetSize()
{
	return size;
}
int Game::Ship::GetFieldValue()
{
	return fieldValue;
}

File: Battleship 25032020/Game/Core/Ship.cs
namespace Game.Core
{
	class Ship
	{
		private string name;
		private int size;
		private int fieldValueHorizontal;
		private int fieldValueVertical;

		public string Name
		{
			get { return name; }
			private set { name = value; }
		}
		public int Size
		{
			get { return size; }
			private set { size = value; }
		}
		public int FieldValueHorizontal
		{
			get { return fieldValueHorizontal; }
			private set { fieldValueHorizontal = value; }
		}
		public int FieldValueVertical
		{
			get { return fieldValueVertical; }
			private set { fieldValueVertical = value; }
		}

		public Ship(string name, int size, int fieldValueHorizontal, int fieldValueVertical)
		{
			Name = name;
			Size = size;
			FieldValueHorizontal = fieldValueHorizontal;
			FieldValueVertical = fieldValueVertical;
		}
	};
}